### DISCUSSION TITLE
Game loading x2 longer

### DISCUSSION THREAD

**Discussion by kot9ka** [AmProsius/gothic-1-community-patch#379] [General]
With CP 1.2 game load x2 longer the usual. What wrong with my game? I didn't find anything like this in the FAQ, so I'm asking.

**Comment by szapp** [comment]
Thanks for the report. Could you tell us what part of the loading takes longer than expected? i.e. initial game start up until reaching the game menu, loading a saved game from the main menu, or loading a saved a game from a running game. 

There are different processes involved in each of them and I am sure there is room for optimization in some of them.

**Comment by kot9ka** [comment]
While loading the save. I used to not pay attention, but when I started to save-load very often (on death from the enemy), it began to annoy me. I temporarily disabled the patch and noticed that loading the save is much faster.
I am not sure about x2 but it very noticeble, so maybe 1.5-2 time longer. Maybe more, not sure.

**Comment by szapp** [comment]
Thanks. Is this in the default world? There are some fixes that are reverted on saving and reapplied on every load (depending on the current world) in order to preserve untainted game saves without any traces of the patch. The time each of these rather few revertible fixes is measured. That will provide insights into which fixes may pose a bottleneck.

I can provide instructions on how to enable these measurements maybe tomorrow when I find the time. 

In the meantime, could you provide info on how you start your game? I.e. Spine (player or developer setting), Steam, Steam+Spine, manual retail installation (GothicStarter or GothicStarter_mod)

**Comment by kot9ka** [comment]
Installation order:
Gothic 1.08k (CD version) + 1.08k Player Kit + System Pack 1.8 + Ninja (last build) + Community Patch (last build)
Started the game through GothicStarter.exe

My trash PC (dxdiag):
 Processor: AMD Athlon(tm) II X3 435 Processor (3 CPUs), ~2.9GHz
 Memory: 6144MB RAM
 Available OS Memory: 6140MB RAM
 Card name: AMD Radeon HD 6570



I think there is no need to do a trace. I used the method of the camera. I filmed the loading time of both original and Community Patch. Here are the results:
original: st3313 et5740  (diff 2427ms)
community patch: st2437 et6587 (diff 4150ms)
4150/2427=1.7
So I was very close. It's x 1.7 longer.
(save loade on chapter 2 right before InExtremo concert, but this happens on every save in the big world)

I want to ask what is the normal extra time fo this patch? On your PC for example,

**Comment by szapp** [comment]
Thanks for the detailed description! That is quite helpful.

You might be right that it is related to the old hardware (in particular the CPU). Up until recently I used an old laptop for development on my end to spot optimization problems. Yet, there we made sure the delay would stay well below 500 ms. With this margin we expected no noticeable delay on current hardware. 

Nevertheless, I will pick this up into our list and we will try to optimize these fixes in the future. Once we get there, may I contact you here to help us identify the slowest of these fixes?

**Reply by kot9ka** [reply]
Yes. You can contact me.  I will try.

**Comment by jgadipriss** [comment]
freezes at the entrance, a black screen is visible and then there is no progress

**Reply by szapp** [reply]
It seems that this might be an unrelated issue. Please elaborate your problem with a few more details - preferably in a new, separate discussion topic.
